pkg/api/types.go

```diff
@@ -799,6 +799,22 @@ type NodeCondition struct {
 	Message            string            `json:"message,omitempty"`
 }
 
+type NodeAddressKind string
+
+// These are valid address type of node. NodeLegacyHostIP is used to transit from out-dated
+// HostIP field to NodeAddress.
+const (
+	NodeLegacyHostIP NodeAddressKind = "LegacyHostIP"
+	NodeHostName     NodeAddressKind = "Hostname"
+	NodeExternalIPv4 NodeAddressKind = "ExternalIPv4"
```

Comment:
Why specify v4? We can distinguish v4 vs. v6, if necessary, by looking at the address.